packages/databases-collections/src/components/drop-collection-modal.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useCallback } from 'react';
+import { connect } from 'react-redux';
+import {
+  Banner,
+  Body,
+  css,
+  FormFieldContainer,
+  FormModal,
+  spacing,
+  SpinLoader,
+  TextInput
+} from '@mongodb-js/compass-components';
+
+import { changeCollectionName } from '../modules/drop-collection/name';
+import { changeCollectionNameConfirmation } from '../modules/drop-collection/name-confirmation';
+import { dropCollection } from '../modules/drop-collection';
+import { toggleIsVisible } from '../modules/is-visible';
+
+const progressContainerStyles = css({
+  display: 'flex',
+  gap: spacing[2],
+  alignItems: 'center',
+});
+
+type DropCollectionModalProps = {
+  isRunning: boolean;
+  isVisible: boolean;
+  name: string;
+  nameConfirmation: string;
+  error?: Error;
+  changeCollectionNameConfirmation: (name: string) => void;
+  dropCollection: () => void;
+  toggleIsVisible: (isVisible: boolean) => void;
+};
+
+function DropCollectionModal({
+  isRunning,
+  isVisible,
+  name,
+  nameConfirmation,
+  error,
+  changeCollectionNameConfirmation,
+  dropCollection,
+  toggleIsVisible,
+}: DropCollectionModalProps) {
+  const onNameConfirmationChange = useCallback((evt: React.ChangeEvent<HTMLInputElement>) => {
+    changeCollectionNameConfirmation(evt.target.value);
+  }, [ changeCollectionNameConfirmation ]);
```

Comment:
Maybe a dumb question, but would it make sense to move this into the UI state along the lines of `const [ nameConfirmation, changeCollectionNameConfirmation ] = useState('');` (and similarly in the drop-datbase-modal)? Obviously not something that’s really being introduced here but it would seem like something that makes sense to me, since the confirmation string is really just affecting UI state.

Reply:
Makes sense to me, those are better in the state for a component like this. How they are currently in the store does make it so if the component unmounts and is remounted it can persist the name confirmation. I don't see how that would help for these components thought. Updating 👍